include/mbedtls/ecdh.h

```diff
@@ -70,7 +116,26 @@ typedef struct mbedtls_ecdh_context
 #if defined(MBEDTLS_ECP_RESTARTABLE)
     int restart_enabled;        /*!< The flag for restartable mode. */
     mbedtls_ecp_restart_ctx rs; /*!< The restart context for EC computations. */
-#endif
+#endif /* MBEDTLS_ECP_RESTARTABLE */
+#else
+    uint8_t point_format;       /*!< The format of point export in TLS messages
+                                  as defined in RFC 4492. */
+    mbedtls_ecp_group_id grp_id;/*!< The elliptic curve used. */
+    mbedtls_ecdh_variant var;   /*!< The ECDH implementation/structure used. */
+    union
+    {
+        mbedtls_ecdh_context_mbed   mbed_ecdh;
+    } ctx;                      /*!< Implementation-specific context. The
+                                  context in use is specified by the \c var
+                                  field. */
+#if defined(MBEDTLS_ECP_RESTARTABLE)
+    uint8_t restart_enabled;    /*!< The flag for restartable mode. Functions of
+                                  an alternative implementation not supporting
+                                  restartable mode must return
+                                  MBEDTLS_ERR_ECP_FEATURE_UNAVAILABLE error if
```

Comment:
The error should be `MBEDTLS_ERR_PLATFROM_FEATURE_UNSUPPORTED` once #2054 will be merged

Reply:
Also, in #2165 we agreed that an alternative implementation and the restartable feature should not co exist, as they don't make sense to be together. We also added protection for that in `check_config.h`
We should consider this in the design of this PR

Reply:
About the error: that is a good point, this needs to be fixed in whichever PR gets merged later. I suggest we make a note of this at #2054 too.

About the restartable feature: This PR aims for a tighter integration than #2165 and unlike there, here we have a sensible way of making them coexist. I think in some scenarios it makes perfect sense to use them together. The alternative implementation might be restartable too and even if it is not, it still might be so much faster that it does not need it. Of course a typical alternative implementation will not implement all of the curves Mbed TLS supports.